packages/connection-form/src/utils/csfle-handler.ts

```diff
@@ -0,0 +1,199 @@
+  connectionOptions: ConnectionOptions;
+}): {
+  connectionOptions: ConnectionOptions;
+} {
+  connectionOptions = cloneDeep(connectionOptions);
+  const autoEncryption = connectionOptions.fleOptions?.autoEncryption ?? {};
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  let tls: any = { ...(autoEncryption.tlsOptions?.[action.kms] ?? {}) };
+  if (!action.value) {
+    delete tls[action.key];
+  } else {
+    tls[action.key] = action.value;
+  }
+  if (Object.keys(tls).length === 0) {
+    tls = undefined;
+  }
+  return {
+    connectionOptions: {
+      ...connectionOptions,
+      fleOptions: {
+        storeCredentials: false,
+        ...connectionOptions.fleOptions,
+        autoEncryption: {
+          ...autoEncryption,
+          tlsOptions: {
+            ...autoEncryption.tlsOptions,
+            [action.kms]: tls,
+          },
+        },
+      },
+    },
+  };
+}
+
+export function hasAnyCsfleOption(o: Readonly<AutoEncryptionOptions>): boolean {
+  return !!(
+    o.bypassAutoEncryption ||
+    o.keyVaultNamespace ||
+    o.schemaMap /* TODO(COMPASS-5645): encryptedFieldConfig */ ||
+    [
+      ...Object.values(o.tlsOptions ?? {}),
+      ...Object.values(o.kmsProviders ?? {}),
+    ]
+      .flatMap((o) => Object.values(o))
+      .filter(Boolean).length > 0
+  );
+}
+
+export function textToEncryptedFieldConfig(text: string): Document {
+  // We use `$compass`-prefixed strings here since the keys of
+  // `encryptedFieldConfigMap` refer to databases and
+  // those never contain `$`.
+
+  let parsed: Document = {};
+  try {
+    parsed = queryParser(text);
+    parsed['$compass.error'] = null;
+  } catch (err: unknown) {
+    parsed['$compass.error'] = (err as Error).message;
+  }
+
+  parsed['$compass.rawText'] = text;
+  return parsed;
+}
+
+export function encryptedFieldConfigToText(config: Document): string {
+  if (config['$compass.rawText']) {
+    return config['$compass.rawText'];
+  }
+  const withoutCompassKeys = Object.fromEntries(
+    Object.entries(config).filter(([key]) => !key.startsWith('$compass.'))
+  );
+  return queryParser.toJSString(withoutCompassKeys);
+}
+
+// The CSFLE encryptedFieldConfigMap contains BSON values, including
+// UUIDs, which are not serialized correctly by the connection model.
+// To account for this, we regenerate the actual values from its
+// text representation on connection form load.
+export function adjustCSFLEParams(
+  connectionOptions: Readonly<ConnectionOptions>
+): ConnectionOptions {
+  connectionOptions = cloneDeep(connectionOptions);
+  const autoEncryptionOptions = connectionOptions.fleOptions?.autoEncryption;
+  // TODO(COMPASS-5645): schemaMap -> encryptedFieldConfig[Map?]
+  if (autoEncryptionOptions?.schemaMap?.['$compass.error'] === null) {
+    autoEncryptionOptions.schemaMap = textToEncryptedFieldConfig(
+      autoEncryptionOptions.schemaMap['$compass.rawText']
+    );
+  }
+  return connectionOptions;
+}
```

Comment:
@mabaasit Fwiw, the last three functions here are the solution that I came up with w.r.t. what we chatted about today -- this solution here has the advantages that the driver options are always in a usable state and we do keep the actual editor contents around without needing an extra top-level field for storage, if you have any thoughts, feel free to share them :)